packages/data-stores/src/queries/use-all-domains-query.ts

```diff
@@ -22,10 +23,30 @@ export interface AllDomainsQueryFnData {
 }
 
 export function useAllDomainsQuery( options: UseQueryOptions< AllDomainsQueryFnData > = {} ) {
+	const testDataParam = getTestDataParam();
+
 	return useQuery( {
-		queryKey: [ 'all-domains' ],
+		queryKey: [ 'all-domains', testDataParam ],
 		queryFn: () =>
-			wpcomRequest< AllDomainsQueryFnData >( { path: '/all-domains', apiVersion: '1.1' } ),
+			wpcomRequest< AllDomainsQueryFnData >( {
+				path: addQueryArgs( '/all-domains', testDataParam ),
+				apiVersion: '1.1',
+			} ),
 		...options,
 	} );
 }
+
+function getTestDataParam() {
+	const testData = typeof window !== 'undefined' && window.localStorage.getItem( 'test-domains' );
```

Comment:
based on your comment in D119028-code

> I see how we need both endpoints to be mocked. But could we change this conditional to defined( 'USE_MOCKED_DOMAINS_LIST' ) && USE_MOCKED_DOMAINS_LIST so that both endpoints use the same flag?

I used the constant so there's no need for the `test_data`  query param, only the `domain_count` to determine the number of mocks to return

Reply:
Also @zaguiini suggest using a constant to set the domain count on the sandbox also. I think that make sense.